app/src/main/java/com/hedvig/app/feature/embark/passages/textactionset/TextActionFragment.kt

```diff
@@ -90,19 +95,61 @@ class TextActionSetFragment : Fragment(R.layout.fragment_text_action_set) {
         delay(PASSAGE_ANIMATION_DELAY_MILLIS)
     }
 
-    private fun textFieldData(data: TextActionSetParameter) =
-        data.keys.mapIndexed { index, key ->
-            TextFieldData(
-                key,
-                data.placeholders[index],
-                data.mask[index],
-                key?.let { model.getFromStore(it) },
-            )
+    private fun createInputViews(): List<View> {
+        return data.keys.mapIndexed { index, key ->
```

Comment:
I'm partial to the functional syntax for method declaration, in particular when the first statement in a method is a return-statement:

```suggestion
    private fun createInputViews() = data.keys.mapIndexed { index, key ->
```

(note that merging this suggestion requires one curly bracket to be removed at the end of the method 😉 )